code/xcvm/lib/cow/src/msg.rs

```diff
@@ -0,0 +1,106 @@
+use alloc::string::String;
+use bounded_collections::{BoundedVec, ConstU32};
+use cosmwasm_std::{Addr, Coin, CosmosMsg, Uint64};
+use xcvm_core::{
+	AssetId, CallOrigin, DefaultXCVMPacket, Instruction, Network, Program, UserOrigin,
+};
+
+enum Decision {
+	Approve{solution_id : Option<String> },
+	/// if there was solution, users lists it lock
+	/// if no solution yet, he get all back to prevent ddos solvers
+	Cancel,
+	Reject,
+}
+
+
+pub enum ExecuteMsg {
+	SubmitIntention { intention: BoundedVec<Intention, ConstU32<4>> },	
```

Comment:
Why bounded? In principle this can be as big as one desires. Similarly with other places.